components/credential-task/credential-task-directive.js

```diff
@@ -157,6 +173,26 @@ function brCredentialTaskDirective() {
       }
       return _.isEqual(query, CRYPTO_KEY_REQUEST);
     }
+
+    function _makePublic(credentials) {
+      var updatePromises = [];
+      for(var index in credentials) {
```

Comment:
Don't use `for(var i in x)` for arrays, use `for(var i = 0; i < x.length; ++i)` or some iterator function.